### PR TITLE
Use ignition provisioning

### PR DESCRIPTION
**Pull request by corest**

### CODE REVIEW

**Review by calvix — APPROVED** [review]
lgtm, maybe we can link the `qemu-node-setup` as I would like to see the code and where its located

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -144,9 +143,23 @@ fi
...
 #
 
-mkdir -p "${raw_cloud_config_dir}"
-cat "${CLOUD_CONFIG_PATH}" | base64 -d | gunzip > "${raw_cloud_config_path}"
-echo "hostname: '${HOSTNAME}'" >> "${raw_cloud_config_path}"
+# Extract ignition from mounted configmap into raw provision config.
+cat "${CLOUD_CONFIG_PATH}" | base64 -d | gunzip > "${raw_ignition_dir}/${ROLE}.json"
+
+# Generate final ignition with static network configuration and hostname
+# Usage of ./qeme-node-setup:
```
Comment (xh3b4sd) [inline docker-entrypoint.sh]:
```suggestion
# Usage of ./qemu-node-setup:
```

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -144,9 +143,23 @@ fi
 # Boot the VM.
 #
 
-mkdir -p "${raw_cloud_config_dir}"
-cat "${CLOUD_CONFIG_PATH}" | base64 -d | gunzip > "${raw_cloud_config_path}"
-echo "hostname: '${HOSTNAME}'" >> "${raw_cloud_config_path}"
+# Extract ignition from mounted configmap into raw provision config.
+cat "${CLOUD_CONFIG_PATH}" | base64 -d | gunzip > "${raw_ignition_dir}/${ROLE}.json"
+
+# Generate final ignition with static network configuration and hostname
```
Comment (xh3b4sd) [inline docker-entrypoint.sh]:
```suggestion
# Generate final ignition with static network configuration and hostname.
#
```

File: start.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
```
Comment (xh3b4sd) [inline start.sh]: What is this now used for? Why do we have `start.sh` and `docker-entrypoint.sh`?

**Review by xh3b4sd — COMMENTED** [review]
Where does the `qemu-node-setup` binary come from? How can we ensure it does not any nasty things like mining cryptos?

**Review by r7vme — APPROVED** [review]

File: start.sh (hunk 3 above)
Comment (calvix) [reply start.sh]: looks like som test file

File: Dockerfile [hunk 4]
```diff
@@ -1,10 +1,11 @@
-FROM fedora:24
+FROM fedora:29
```
Comment (tuommaki) [inline Dockerfile]: :heart:

**Review by tuommaki — APPROVED** [review]
LGTM once other comments are addressed.

File: start.sh (hunk 3 above)
Comment (corest) [reply start.sh]: Yes, debug file. Removed

**Comment by corest** [comment]
@xh3b4sd qemu-node-setup is here https://github.com/giantswarm/qemu-node-setup . It is small utility I wrote to generate final ignition for some parameters, which we are able to get only in runtime

**Comment by xh3b4sd** [comment]
Let's please document where the binary comes from in the readme so we know what is going on.

**Comment by corest** [comment]
All good here. Tested on gorgoth/geckon as well as on amagon, where we had issues previously with fedora:26 and nested virtualization. `fedora:29` works better

**Pull request merged**